internal/identity/identitymanager.go

```diff
@@ -105,13 +105,13 @@ func ParseKeyNormalizationConfig(strConfigVal string) int {
 
 func (im *identityManager) GetLocalNode(ctx context.Context) (node *core.Identity, err error) {
 	nodeName := im.multiparty.LocalNode().Name
-	if nodeName != "" {
-		nodeDID := fmt.Sprintf("%s%s", core.FireFlyNodeDIDPrefix, nodeName)
-		node, _, err = im.CachedIdentityLookupNilOK(ctx, nodeDID)
-	}
-	if err == nil && node == nil {
+	nodeDID := fmt.Sprintf("%s%s", core.FireFlyNodeDIDPrefix, nodeName)
+	node, _, err = im.CachedIdentityLookupNilOK(ctx, nodeDID)
+
+	if nodeName == "" && node == nil {
```

Comment:
You could simply check if `nodeName==""` and then return the error, if not just return node and err

Reply:
:rocket: